onnxruntime/core/providers/cuda/tensor/eye_like_impl.cu

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include <stdint.h>
+#include "core/providers/cuda/shared_inc/cuda_utils.h"
+#include "core/providers/cuda/cu_inc/common.cuh"
+#include "eye_like_impl.h"
+
+namespace onnxruntime {
+namespace cuda {
+
+template <typename T>
+__global__ void _EyeLikeKernel(
+    const int64_t k,
+    const fast_divmod fdm_x,
+    T* output_data,
+    CUDA_LONG N) {
+  CALCULATE_ELEMENTWISE_INDEX_OR_EXIT(id, N);
+  int x, y;
+  fdm_x.divmod(id, x, y);
+  T value = static_cast<T>(0);
+
+  if ((k >=0 && (x + k) == y) || (k < 0 && (y - k) == x))
```

Comment:
>((k >=0 && (x + k) == y) || (k < 0 && (y - k) == x)) [](start = 5, length = 52)

since both x,y,k are int, the expression is just x+k == y for both k>=0 and k <0?